classes/pdf/HTMLTemplateOrderSlip.php

```diff
@@ -67,11 +67,40 @@ public function __construct(OrderSlip $order_slip, Smarty $smarty)
         // header informations
         $this->date = Tools::displayDate($this->order_slip->date_add);
         $prefix = Configuration::get('PS_CREDIT_SLIP_PREFIX', Context::getContext()->language->id);
-        $this->title = sprintf(HTMLTemplateOrderSlip::l('%1$s%2$06d'), $prefix, (int) $this->order_slip->id);
+
+        if (Configuration::get('PS_CREDIT_SLIP_RESET')) {
+            $this->title = sprintf(HTMLTemplateOrderSlip::l('%1$s%2$06d'), $prefix, $this->getNumberOrderSlip((int) $order_slip->id));
+        } else {
+            $this->title = sprintf(HTMLTemplateOrderSlip::l('%1$s%2$06d'), $prefix, (int) $order_slip->id);
+        }
 
         $this->shop = new Shop((int) $this->order->id_shop);
     }
 
+    /**
+     * Returns the numbering of the subscription invoice with respect to the year.
+     *
+     * @param int $id_order_slip order_slip
+     *
+     * @return int number for year
+     */
+    public function getNumberOrderSlip(int $id_order_slip): int
+    {
+        $row = Db::getInstance()->executeS('SELECT date_add FROM `' . _DB_PREFIX_ . 'order_slip` WHERE id_order_slip=' . $id_order_slip);
```

Comment:
I think you can use `Db::getInstance()->getValue()` instead of `Db::getInstance()->executeS()`

Reply:
OK. Perfect I make the change, you are right that way it is better thanks.